### PR TITLE
Backport forgotten backports [1.1.1]

### PR DESCRIPTION
**Pull request by mspncp**
Looking through the git logs with [cherry-checker](https://github.com/openssl/tools/blob/master/review-tools/cherry-checker) I found some commits which might deserve some backporting to the 1.1.1 stable branch.

 prnum  |   commit   |   backport   |   subject
------- | ---------- | ---------- | -------------------------------------------
 #9094  | 8caab503ba | 0efb388c18 |  Fix an incorrect macro  
 #9094  | 4ff4e53f81 | c00b5ded40 |  i2d_PublicKey was listed in 2 different man pages 
 #8578  | f3448f5481 | 4b3ccc12be |  issue-8493: Fix for filenames with newlines using openssl dgst 
 #8520  | d7b2124a42 | ca0d329cd9 |  Add documentation for the -sigopt option. 
 #8093  | a97faad76a | 3c62782110 |  Document and add macros for additional DSA options 
 #7905  | 37842dfaeb | bc9d91aa04 |  Add missing EVP_MD documentation  
 #7423  | ee4afacd96 | f5c5021847 |  Ignore duplicated undocumented things 

Please let me know whether you agree with the list or whether some backports should be added resp. removed.

And please let me also know whether you prefer separate pull requests per backport or would accept this bulk backport in a single pull request.

### CODE REVIEW

**Comment by mspncp** [comment]
_(Note: The optimal way to view this pull request is to go to the 'Files changed' tab and narrow down the scope to single commits, since the commits are unrelated.)_

**Comment by mspncp** [comment]
ping

File: apps/dgst.c [hunk 1]
```diff
@@ -413,13 +413,52 @@ int dgst_main(int argc, char **argv)
...
+ * The newline_escape_filename function performs newline escaping for any
+ * filename that contains a newline.  This function also takes a pointer
+ * to backslash. The backslash pointer is a flag to indicating whether a newline
+ * is present in the filename.  If a newline is present, the backslash flag is
+ * set and the output format will contain a backslash at the beginning of the
+ * digest output. This output format is to replicate the output format found
+ * in the '*sum' checksum programs. This aims to preserve backward
+ * compatibility.
+ */
+static const char *newline_escape_filename(const char *file, int * backslash)
```
Comment (mattcaswell) [inline apps/dgst.c]: Too many spaces `int * backslash`

File: apps/dgst.c [hunk 2]
```diff
@@ -413,13 +413,52 @@ int dgst_main(int argc, char **argv)
...
+
+    for (i = 0; i < length; i++)
+        if (file[i] == '\n')
+            newline_count++;
+
+    mem_len = length + newline_count + 1;
+    file_cpy = app_malloc(mem_len, file);
+    i = 0;
+
+    while(e < length) {
```
Comment (mattcaswell) [inline apps/dgst.c]: Missing space after while

File: crypto/dsa/dsa_pmeth.c [hunk 3]
```diff
@@ -178,9 +178,7 @@ static int pkey_dsa_ctrl_str(EVP_PKEY_CTX *ctx,
     }
     if (strcmp(type, "dsa_paramgen_q_bits") == 0) {
         int qbits = atoi(value);
-        return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_DSA, EVP_PKEY_OP_PARAMGEN,
-                                 EVP_PKEY_CTRL_DSA_PARAMGEN_Q_BITS, qbits,
-                                 NULL);
+        return EVP_PKEY_CTX_set_dsa_paramgen_q_bits(ctx, qbits);
```
Comment (mattcaswell) [inline crypto/dsa/dsa_pmeth.c]: I am unsure about backporting this macro.

Comment (levitte) [reply crypto/dsa/dsa_pmeth.c]: Considering the text of the commit, that this is a programming safety measure, I think it's ok.

Comment (mspncp) [reply crypto/dsa/dsa_pmeth.c]: A missing ctrl wrapper macro for a ctrl code could also be considered to be analogous to a missing accessor. But as I indicated in the description, the list of backports is debatable. If Matt prefers not to have it backported, I could drop it.

Comment (mattcaswell) [reply crypto/dsa/dsa_pmeth.c]: I'm ok with it. I just had a moments hesitation as to whether we were overstepping our rules with this change.

**Comment by mspncp** [comment]
Pushed to 1.1.1, thanks!

7bcd13cebd Fix an incorrect macro
abf92a9715 i2d_PublicKey was listed in 2 different man pages
109a00269d issue-8493: Fix for filenames with newlines using openssl dgst
3c682fad5f Add documentation for the -sigopt option.
44301079c8 Document and add macros for additional DSA options
0388d212af Add missing EVP_MD documentation
3fb4bdabc2 Ignore duplicated undocumented things

(@mattcaswell I'll take care of the whitespace issues later)

**Pull request closed without merging**